commands/command_edit.cpp

```diff
@@ -1285,6 +1286,58 @@ void EditCylinder::undo()
 	redo();
 }
 
+EditSensors::EditSensors(int cylIndex, divecomputer *dcIn, std::vector<int> sensors)
+	: EditDivesBase(true), dc(dcIn)
+{
+	if (dives.empty())
+		return;
+
+	setText(Command::Base::tr("Edit sensors"));
+
+	for (int s = 0; s < MAX_SENSORS; ++s) {
+		if (std::find(sensors.begin(), sensors.end(), s) != sensors.end()) {
+			// If this sensor is in the list of sensors being added to this cylinder
+			// then store the old cylinder in undoMapping and the new cylinder in redoMapping.
+			undoMapping.emplace_back(s, dc->sample[0].sensor[s]);
+			redoMapping.emplace_back(s, cylIndex);
+		} else if (dc->sample[0].sensor[s] == cylIndex) {
+			// Otherwise, if the sensor was previously attached to this cylinder then
+			// save that in the undoMapping and set it to NO_SENSOR in redoMapping so
+			// it is not attached to any cylinder.
+			undoMapping.emplace_back(s, cylIndex);
+			redoMapping.emplace_back(s, NO_SENSOR);
+		}
+	}
+
+	mapSensors(redoMapping);
+}
+
+void EditSensors::mapSensors(std::vector<std::pair<int, int>> &mapping)
+{
+	for (int i = 0; i < dc->samples; ++i) {
+		for (auto &[s, cyl] : mapping) {
+			dc->sample[i].sensor[s] = cyl;
+		}
+	}
+	emit diveListNotifier.deviceEdited();
```

Comment:
I wonder if `cylindersReset` would be more appropriate. Or a new signal?

Reply:
`cylindersReset` doesn't replot the profile, which is what I needed to emit the signal for. Adding a diveComputerEdited signal, then it could be changed so the replotting isn't done if you undo the edit while viewing another dive computer, not sure if that change would be worth the added complexity though.